### PR TITLE
Delete medical litter on building explosion

### LINKED ISSUE #5412: floating medical junk
On all versions without no mods and all mods the medical junk does not drop/destroy when a building is destroyed.
Medical junk missing physx

http://i.imgur.com/VjsKqbO.jpg

### PR DESCRIPTION
**Pull request by PabstMirror**
Based on commy's idea
Should fix #5412

### CODE REVIEW

**Comment by SilentSpike** [comment]
Code looks good, will have to test in-game how reliable/effective it is

File: addons/medical/XEH_postInit.sqf [hunk 1]
```diff
@@ -31,6 +31,20 @@ GVAR(heartBeatSounds_Slow) = ["ACE_heartbeat_slow_1", "ACE_heartbeat_slow_2"];
 if (isServer) then {
     ["ace_placedInBodyBag", FUNC(serverRemoveBody)] call CBA_fnc_addEventHandler;
     [QGVAR(createLitterServer), FUNC(handleCreateLitter)] call CBA_fnc_addEventHandler;
+    addMissionEventHandler ["BuildingChanged", {
+        if (isNil QGVAR(allCreatedLitter)) exitWith {};
+        params ["_buildingOld", "_buildingNew", "_isRuin"];
```
Comment (thojkooi) [inline addons/medical/XEH_postInit.sqf]: Shouldn't `_isRuin` need to be checked before cleaning up?

File: addons/medical/XEH_postInit.sqf [hunk 2]
```diff
@@ -31,6 +31,20 @@ GVAR(heartBeatSounds_Slow) = ["ACE_heartbeat_slow_1", "ACE_heartbeat_slow_2"];
...
+        TRACE_3("BuildingChanged",_buildingOld,_buildingNew,_isRuin);
+        private _radius = ((0.75 * getNumber (configFile >> "CfgVehicles" >> (typeOf _buildingOld) >> "mapsize")) max 5) min 50;
```
Comment (thojkooi) [inline addons/medical/XEH_postInit.sqf]: What's the `50` based on? Seems like a rather large and specific number for some buildings.

File: addons/medical/XEH_postInit.sqf [hunk 3]
```diff
@@ -31,6 +31,20 @@ GVAR(heartBeatSounds_Slow) = ["ACE_heartbeat_slow_1", "ACE_heartbeat_slow_2"];
...
+        TRACE_1("",_radius);
+        {
+            _x params ["", "_objects"];
+            if (({(_x distance2d _buildingOld) < _radius} count _objects) > 0) then {
```
Comment (thojkooi) [inline addons/medical/XEH_postInit.sqf]: This also cleans up anything that is not inside the building. Perhaps check for anything that's up in the air?

File: addons/medical/XEH_postInit.sqf (hunk 2 above)
Comment (SilentSpike) [reply addons/medical/XEH_postInit.sqf]: I'd suggest we just use the largest bounding box dimension

File: addons/medical/XEH_postInit.sqf (hunk 3 above)
Comment (SilentSpike) [reply addons/medical/XEH_postInit.sqf]: Collateral damage on valid medical litter isn't very important, but a quick height check isn't a bad idea 👍

File: addons/medical/XEH_postInit.sqf (hunk 2 above)
Comment (commy2) [reply addons/medical/XEH_postInit.sqf]: sizeOf typeOf _buildingOld

File: addons/medical/XEH_postInit.sqf (hunk 3 above)
Comment (commy2) [reply addons/medical/XEH_postInit.sqf]: `getPos select 2` is perfect for this. Reports the height above the first roadway LOD /or the terrain.

File: addons/medical/XEH_postInit.sqf (hunk 1 above)
Comment (commy2) [reply addons/medical/XEH_postInit.sqf]: Some buildings might change appearance enough to make some things float before they are ruins.

**Pull request merged**